### PR TITLE
update to 0.21.2 rebuild with vtk 9.2.6, hdf5 1.14.2 and libbosst 1.82

### PR DESCRIPTION
**Pull request by adrianinsaval**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -69,7 +69,7 @@ requirements:
         - qt
         - xerces-c
         - zlib
-        - boost-cpp
+        - libboost
```
Comment (h-vetinari) [inline recipe/meta.yaml:72]: This isn't necessary; `libboost-devel` will ensure that the right `libboost` is available at runtime through a so-called run-export.

**Comment by oursland** [comment]
The `occt==7.6.3` pinning is preventing successful resolution.

**Comment by adrianinsaval** [comment]
I can remove the pin but freecad has known bugs with occt 7.7 so I'm not sure if it's worth it. I think the only thing I'm missing to get it working with occt 7.6.3 is gmsh. I could remove gmsh from here I think, if I'm not mistaken is not actually required during build. It's an optional runtime dependency for the FEM module

**Comment by oursland** [comment]
I can confirm that it builds and runs successfully with occt 7.7.2.  If there are bugs, that would be quite detrimental, though.

**Comment by adrianinsaval** [comment]
there's at least https://github.com/FreeCAD/FreeCAD/issues/9651 and suspected https://github.com/FreeCAD/FreeCAD/issues/9760

**Comment by looooo** [comment]
I would suggest to build for both (occt 7.6.3 and occt 7.7.x) I guess you can add variants here, but I am not sure:
https://github.com/conda-forge/freecad-feedstock/blob/main/recipe/conda_build_config.yaml

**Comment by adrianinsaval** [comment]
these errors are bogus, what's up with conda not being able to setup the test environment?

**Pull request merged**